helm/v1beta1/types_helm.go

```diff
@@ -0,0 +1,87 @@
+package v1beta1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:plural=helmchartrepositories
+
+// HelmChartRepository holds cluster-wide configuration for proxied Helm chart repository
+type HelmChartRepository struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec HelmChartRepositorySpec `json:"spec"`
+
+	// Observed status of the repository within the cluster..
+	// +optional
+	Status HelmChartRepositoryStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type HelmChartRepositoryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []HelmChartRepository `json:"items"`
+}
+
+// Helm chart repository exposed within the cluster
+type HelmChartRepositorySpec struct {
+
+	// Optional associated human readable repository name, it can be used by UI for displaying purposes
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	DisplayName string `json:"name,omitempty"`
+
+	// Optional human readable repository description, it can be used by UI for displaying purposes
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Required configuration for connecting to the chart repo
+	ConnectionConfig `json:",inline"`
+}
+
+type ConnectionConfig struct {
+
+	// Chart repository URL
+	// +kubebuilder:validation:Pattern=`^https?:\/\/`
+	URL string `json:"url"`
+
+	// ca is an optional reference to a config map by name containing the PEM-encoded CA bundle.
+	// It is used as a trust anchor to validate the TLS certificate presented by the remote server.
+	// The key "ca.crt" is used to locate the data.
+	// If empty, the default system roots are used.
+	// The namespace for this config map is openshift-config.
+	// +optional
+	CA *configv1.ConfigMapNameReference `json:"ca,omitempty"`
+
+	// tlsClientCert is an optional reference to a secret by name that contains the
+	// PEM-encoded TLS client certificate to present when connecting to the server.
+	// The key "client.crt" is used to locate the data.
+	// The namespace for this secret is openshift-config.
+	// +optional
+	TLSClientCert *configv1.SecretNameReference `json:"tlsClientCert,omitempty"`
```

Comment:
only one reference for both cert and key. follow the kube tls secret type for your keys. This means `tls.crt` and `tls.key`

Reply:
ok, I have initially modelled after https://github.com/openshift/api/blob/99797c0ac2b5ef6275b1108e9c507d3eaab53246/config/v1/types_oauth.go#L263